humblegen/tests/rust/service/spec.humble

```diff
@@ -0,0 +1,83 @@
+// We are going to define the service and its data types and subsequently
+// explain the philosophy behind the idea.
+
+
+/// A wandering monster
+struct Monster {
+    /// Monster ID.
+    id: i32,
+    .. MonsterData,
+    .. MonsterData2,
+}
+
+struct MonsterData {
+    /// The monster's name
+    name: str,
+    /// Max hitpoints.
+    hp: i32,
+}
+
+struct MonsterData2 {
+    foo: str,
+    .. MonsterData3,
+}
+
+struct MonsterData3 {
```

Comment:
At some point, we should separate the "educational" specs from those testing the parser/generator, as this is mighty confusing...